### PR TITLE
Products page

### PR DESCRIPTION
**Pull request by epashiq**

### CODE REVIEW

File: lib/core/constants/products_constants/product_constants.dart [hunk 1]
```diff
@@ -5,6 +5,27 @@ part 'product_constants.g.dart';
...
+  final txtFullQty = 'Full';
+  final txtFullQtyPrice = '\$25';
+  final txtHalfQty = 'Half';
+  final txtHalfQtyPrice ='\$15';
+  final elevatedBtnTxt = 'Add Type';
+  final txtAddOns = 'Add-Ons';
+  final txtAddCategoryTitle = 'Add New Category';
+  final txtAddOnBtnTitle = 'Add Add-Ons';
+  final txtSaveBtn = 'Save';
+  final addImgPath = 'assets/icons/ic_add_image.svg';
```
Comment (thadathilsinan) [inline lib/core/constants/products_constants/product_constants.dart]: Addd into constants

File: lib/core/widgets/elevated_add_button_widget.dart [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      {super.key,
+      required this.buttonText,
+      required this.textColor,
+      required this.onPressed,
+      required this.icon});
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      width: MediaQuery.sizeOf(context).width,
+      height: MediaQuery.sizeOf(context).height / 18,
```
Comment (thadathilsinan) [inline lib/core/widgets/elevated_add_button_widget.dart]: Use spaces

File: lib/core/widgets/elevated_add_button_widget.dart [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+              icon,
+              color: AppTheme.of(context).colors.primary,
+              size: AppTheme.of(context).spaces.space_250,
+            ),
+            SizedBox(
+              width: AppTheme.of(context).spaces.space_100,
+            ),
+            Text(
+              buttonText,
+              style: TextStyle(color: textColor, fontFamily: 'inter'),
```
Comment (thadathilsinan) [inline lib/core/widgets/elevated_add_button_widget.dart]: use typography theme

File: lib/core/widgets/image_picker_widget.dart [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+class ImagePickerWidget extends ConsumerWidget {
+  const ImagePickerWidget({super.key});
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final data = ref.watch(productConstantsProvider);
+    return InkWell(
+      onTap: () {},
+      child: Container(
+        height: MediaQuery.sizeOf(context).height / 3.55,
```
Comment (thadathilsinan) [inline lib/core/widgets/image_picker_widget.dart]: use spaces

File: lib/core/widgets/image_picker_widget.dart [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        width: MediaQuery.sizeOf(context).width,
+        decoration: BoxDecoration(
+            borderRadius:
+                BorderRadius.circular(AppTheme.of(context).spaces.space_100),
+            border: Border.all(
+                color: AppTheme.of(context).colors.textSubtle, width: 2)),
```
Comment (thadathilsinan) [inline lib/core/widgets/image_picker_widget.dart]: use spaces

File: lib/core/widgets/image_picker_widget.dart [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+            SizedBox(
+              height: AppTheme.of(context).spaces.space_100,
+            ),
+            Text(
+              data.txtAddImg,
+              style: AppTheme.of(context).typography.h400.copyWith(
+                  color: AppTheme.of(context)
+                      .colors
+                      .textSubtlest
+                      .withOpacity(.60)),
```
Comment (thadathilsinan) [inline lib/core/widgets/image_picker_widget.dart]: NO opacity

File: lib/core/widgets/text_field_widget.dart [hunk 7]
```diff
@@ -3,23 +3,22 @@ import 'package:flutter_riverpod/flutter_riverpod.dart';
...
 
 class TextFieldWidget extends ConsumerWidget {
-  final String textFieldTitle;
+  final String textFielTitle;
   final String hintText;
   final TextEditingController controller;
   const TextFieldWidget(
       {super.key,
-      required this.textFieldTitle,
+      required this.textFielTitle,
```
Comment (thadathilsinan) [inline lib/core/widgets/text_field_widget.dart]: spelling mistake

File: lib/features/edit_profile_page/presentation/pages/edit_password_page.dart [hunk 8]
```diff
@@ -51,17 +51,17 @@ class EditPasswordPage extends HookConsumerWidget {
           child: Column(
             children: [
               TextFieldWidget(
-                  textFieldTitle: constants.txtCurrentPassword,
+                  textFielTitle: constants.txtCurrentPassword,
```
Comment (thadathilsinan) [inline lib/features/edit_profile_page/presentation/pages/edit_password_page.dart]: Spelling mistake

File: lib/features/edit_profile_page/presentation/pages/edit_profile_page.dart [hunk 9]
```diff
@@ -59,12 +59,12 @@ class EditProfilePage extends HookConsumerWidget {
                 ),
                 const SizedBox32Widget(),
                 TextFieldWidget(
-                    textFieldTitle: constants.txtOpeningTime,
```
Comment (thadathilsinan) [inline lib/features/edit_profile_page/presentation/pages/edit_profile_page.dart]: spelling

File: lib/features/products/presentation/pages/add_category_page.dart [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    final apptheme = AppTheme.of(context);
+    final data = ref.watch(productConstantsProvider);
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Row(
+          children: [
+            IconButton(
+                onPressed: () {},
+                icon: SvgPicture.asset('assets/icons/ic_arrow_backward.svg')),
```
Comment (thadathilsinan) [inline lib/features/products/presentation/pages/add_category_page.dart]: use assets const

File: lib/features/products/presentation/pages/product_page.dart [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+    return GestureDetector(
+      onTap: () => FocusScope.of(context).unfocus(),
+      child: Scaffold(
+        appBar: AppBar(
+          scrolledUnderElevation: 0,
+          title: Row(
+            children: [
+              IconButton(
+                  onPressed: () {},
+                  icon: SvgPicture.asset('assets/icons/ic_arrow_backward.svg')),
```
Comment (thadathilsinan) [inline lib/features/products/presentation/pages/product_page.dart]: use asset const

File: lib/features/products/presentation/widgetts/product_type_widget.dart [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        for (int i = 0; i < 2; i++)
+          Padding(
+            padding: EdgeInsets.only(top: apptheme.spaces.space_200),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                SizedBox(
```
Comment (thadathilsinan) [inline lib/features/products/presentation/widgetts/product_type_widget.dart]: use expanded

File: lib/features/products/presentation/widgetts/row_widget.dart [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+  const RowWidget({super.key, required this.text, required this.btnText});
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final theme = AppTheme.of(context);
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+      children: [
+        Text(
+          text,
+          style: const TextStyle(
```
Comment (thadathilsinan) [inline lib/features/products/presentation/widgetts/row_widget.dart]: use theme fonts

File: pubspec.lock
Comment (thadathilsinan) [inline pubspec.lock:1]: Remove this file

**Review by thadathilsinan — CHANGES_REQUESTED** [review]

**Review by Sireen-Official — APPROVED** [review]
Review approved

**Review by Sireen-Official — APPROVED** [review]
Review approved

**Review by thadathilsinan — APPROVED** [review]

**Pull request merged**